smallrye-reactive-messaging-aws-sns/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsConnector.java

```diff
@@ -0,0 +1,214 @@
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.processors.MulticastProcessor;
+import io.reactivex.schedulers.Schedulers;
+import io.vertx.core.Vertx;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.CreateTopicRequest;
+import software.amazon.awssdk.services.sns.model.CreateTopicResponse;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.PublishResponse;
+
+/**
+ * Implement incoming/outcoming connection factories for SNS reactive messaging.
+ * 
+ * @author iabughosh
+ * @version 1.0.4
+ *
+ */
+@ApplicationScoped
+@Connector(SnsConnector.CONNECTOR_NAME)
+public class SnsConnector implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnsConnector.class);
+    static final String CONNECTOR_NAME = "smallrye-aws-sns";
+
+    /*
+     * Injectable fields
+     */
+    @Inject
+    @ConfigProperty(name = "sns-app-public-url")
+    Optional<String> appURL;
+    /** If there is existing verx instance, inject it. */
+    @Inject
+    Instance<Vertx> instanceOfVertx;
+    /*
+     * Fields
+     */
+    /** Control wither close vertx manually or not */
+    private boolean internalVertxInstance = false;
+
+    private Vertx vertx;
+    private ExecutorService threadExecutor;
+    private Scheduler scheduler;
+
+    /**
+     * Method being invoked when CDI bean first created.
+     */
+    @Inject
+    public void initConnector() {
+        //Initialize vertx and threadExecutor
+        LOGGER.info("Initializing Connector");
+        if (instanceOfVertx.isUnsatisfied()) {
+            internalVertxInstance = true;
+            this.vertx = Vertx.vertx();
+        } else {
+            this.vertx = instanceOfVertx.get();
+        }
+        threadExecutor = Executors.newSingleThreadExecutor();
+        scheduler = Schedulers.single();
+    }
+
+    /**
+     * Its being invoked before bean being destroyed.
+     */
+    @PreDestroy
+    public void preDestroy() {
+        LOGGER.info("Destroying Connector");
+        if (internalVertxInstance) {
+            Optional.ofNullable(vertx).ifPresent(vertx -> vertx.close());
+        }
+        Optional.of(threadExecutor).ifPresent(exec -> exec.shutdown());
+        Optional.of(scheduler).ifPresent(sch -> sch.shutdown());
+    }
+
+    @Override
+    public SubscriberBuilder<? extends Message<?>, Void> getSubscriberBuilder(Config config) {
+        //Incase of unit testing only, initialize it.
+        if (threadExecutor == null) {
```

Comment:
I don't understand. Can't you call the `postInit` method in your tests?

Reply:
I think you are right. It didn't cross my mind :(. I will remove it 👍